src/socket/dhcpv4.rs

```diff
@@ -104,32 +99,71 @@ enum ClientState {
     Renewing(RenewState),
 }
 
+/// Timeout and retry configuration.
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub struct RetryConfig {
+    pub discover_timeout: Duration,
+    /// The REQUEST timeout doubles every 2 tries.
+    pub initial_request_timeout: Duration,
+    pub request_retries: u16,
+    pub min_renew_timeout: Duration,
+}
+
+impl RetryConfig {
+    pub const DEFAULT: Self = Self {
```

Comment:
can you change this so there's only the Default impl? otherwise there's 2 ways of doing  the same thing in user code, which is not nice: `RetryConfig::default()` and `RetryConfig::DEFAULT`

Reply:
Had this because I needed a const constructor, but that's no longer the case, so removed.